bigquery/gcloud/aio/bigquery/bigquery.py

```diff
@@ -52,7 +53,7 @@ class SchemaUpdateOption(Enum):
 
 
 class Job:
-    def __init__(self, job_id: str,
+    def __init__(self, job_id: Optional[str] = None,
```

Comment:
Actionable comment: given `job_id` is now `Optional`, we need to reflect that elsewhere. eg. calling `get_job()` on a `Job` with a null ID is gonna do weird string interpolation things and try accessing the job with ID `"None"`. We might need to add some guards to the various methods which check for `if self.job_id is None: raise Exception('cannot get job with no specified ID')` or something? Thoughts?

Reply:
I think I have an old tester's trauma that make me dislike restrictive clients. Oftentimes I _wanted_ to send `resource/None` to a server but a normal production client won't let me. 

Yeah we can raise on that, although that would require a custom exception I believe. Server error is not that bad too `ClientResponseError: 404, message='Not Found: {\n  "error": {\n    "code": 404,\n    "message": "Not found: Job project:None",\n    "errors": [\n      {\n        "message": "Not found: Job project:None",\n        "domain": "global",\n        "reason": "notFound"\n      }\n    ],\n    "status": "NOT_FOUND"\n  }\n}\n', url=URL('https://www.googleapis.com/bigquery/v2/projects/project/jobs/None')` so I'm good either way.

Reply:
Yeah, that's a reasonable enough error message. Alrighty, the whole null-job-id thing looks good to me then!